apps/asap-server/src/controllers/research-outputs.ts

```diff
@@ -138,15 +138,17 @@ export default class ResearchOutputs implements ResearchOutputController {
   async create({
     teamId,
     ...researchOutputData
-  }: ResearchOutputInputData): Promise<Partial<ResearchOutputResponse>> {
+  }: ResearchOutputPostRequest): Promise<Partial<ResearchOutputResponse>> {
```

Comment:
This is the contract between the backend and frontend. It might be remapped after this point but we need to take in a request validate it and then you remap it or whatever.

I find it strange that you treating your controller methods as like the API methods shouldn't there be some separation here?

Reply:
so contract is not at controller level, it is at the request level. This is exactly what i am telling you - to use the `ResearchOutputPostRequest` type like you did inside the RO route but then remap it to something else for the controller. I simply dont want the controller to be tied to the API request contract